kv/replica_slice.go

```diff
@@ -105,7 +107,9 @@ func (rs replicaSlice) SortByCommonAttributePrefix(attrs []string) int {
 	topIndex := len(rs) - 1
 	for bucket := 0; bucket < len(attrs); bucket++ {
 		firstNotOrdered := 0
+		rs.SortRandomOrder(firstNotOrdered)
```

Comment:
`firstNotOrdered` is always zero here. If there is more than one attribute, then each time through the loop we will undo the work from the previous loop. This isn't the right place for this call.